core/drivers/pl022_spi.c

```diff
@@ -157,18 +159,26 @@
 #define SSP_SCR_MIN		0
 #define SSP_DATASIZE_MAX	16
 
+#define SSP_LOOP_TIMEOUT	10000
+
 static void pl022_txrx8(struct spi_chip *chip, uint8_t *wdat, uint8_t *rdat,
 	size_t num_pkts)
 {
 	size_t i = 0;
 	size_t j = 0;
+	size_t cnt = 0;
 	struct pl022_data *pd = container_of(chip, struct pl022_data, chip);
 
 	if (!num_pkts) {
 		EMSG("*num_pkts = 0!\n");
 		return;
 	}
 
+	if (pd->data_size_bits == 16) {
```

Comment:
Wouldn't `if (pd->data_size_bits != 8)` be better? (and same below for the 16-bit version).